[BE][EZ] Migrate to new dcp save and load APIs

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130475

When I play with DCP for distributed inference, I found that we are still using deprecated APIs for DCP even in unit test. So this PR is using the new API with unified small letters "dcp".


cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o @LucasLLC @MeetVadakkanchery @mhorowitz @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @chauhang

fduwjj committed:
[BE][EZ] Migrate to new DCP save and load APIs

[ghstack-poisoned]

wz337 reviewed (approved):
Thanks for updating!

fduwjj commented:
@pytorchbot merge

fduwjj committed:
Update on "[BE][EZ] Migrate to new dcp save and load APIs"


When I play with DCP for distributed inference, I found that we are still using deprecated APIs for DCP even in unit test. So this PR is using the new API with unified small letters "dcp".


cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu fegin XilunWu wanchaol wz337 wconstab chauhang d4l3k LucasLLC MeetVadakkanchery mhorowitz

[ghstack-poisoned]

fduwjj commented:
The new API does not support `no_dist`, we just remove the test cases completely.

fduwjj commented:
@pytorchbot -i merge

fduwjj commented:
@pytorchbot merge -i

fduwjj commented:
Hi, @LucasLLC can you also take a look at this PR to see if the changes look legit to you? Thanks!
